src/pages/index.js

```diff
@@ -4,78 +4,224 @@ import FormValidator from "../components/FormValidator.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, config } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
 import "./index.css";
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
+import Api from "../components/Api.js";
+// import { setLoading } from "../components/PopupWithForm.js";
+
+//APIs
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4a5b23f0-f2a7-4209-a8e7-d3bcf73a20e6",
+    "Content-Type": "application/json",
+  },
+});
+
+export function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardClick,
+    handleDeleteCard,
+    handleLikeCard
+  );
+  return card.getView();
+}
+
+let userInfo;
+let cardSection;
+
+// Load initial cards
+api.loadPage()
+  .then(([initialCards, userData]) => {
+    
+    // Initialize and render the card section
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          const cardElement = renderCard(cardData)
+          cardSection.addItem(cardElement);
+        }
+      },
+      "#cards__list"
+    );
+    cardSection.renderItems();
+
+    // Update user profile
+    userInfo = new UserInfo({
+      nameSelector: ".profile__title",
+      jobSelector: ".profile__description",
+      imageSelector: ".profile__image"
+    });
+    userInfo.setUserInfo(userData);
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+
+// Preview image
+function handleCardClick(name, link) {
+  previewImageModal.open(name, link);
+}
+
+// Add card function
+function handleAddCardSubmit(values) {
+  const name = values.title;
+  const link = values.link;
+    
+  // Set loading state
+  addCardPopup.setLoading(true);
+
+  // Call API method to add a new card
+  api.addNewCard(name, link)
+    .then((newCard) => {
+      // Render the newly created card
+      const cardElement = renderCard(newCard);
+
+      addCardPopup.setLoading(false);
+      addCardPopup.close();
+
+      // Ensure cardSection is accessible and addItem method is defined
+      if (cardSection && cardSection.addItem) {
+        cardSection.addItem(cardElement);
+      } else {
+        console.error('cardSection or addItem method is undefined.');
+      }
+      
+    })
+    .catch((error) => {
+      console.error('Error adding new card:', error);
+      // Reset loading state in case of an error
+    });
+}
+  
+// Update avatar function
+function handleAvatarUpdate(formData) {
+  updateAvatarPopup.setLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```